ESP8266Interface.cpp

```diff
@@ -46,8 +46,16 @@ int ESP8266Interface::connect(const char *ssid, const char *pass, nsapi_security
 
 int ESP8266Interface::connect()
 {
+    _esp.setTimeout(ESP8266_MISC_TIMEOUT);
+    
+    if(!_esp.check_firmware_version()) {
```

Comment:
This needs to occur after the esp8266 is reset (in `ESP8266::startup`) otherwise the esp8266 may be in a bad error state an not respond.